src/util.js

```diff
@@ -0,0 +1,107 @@
+const jsonSchemaFromMSON = require('./mson_to_json_schema')
+const GenerateSchema = require('generate-schema')
+
+module.exports.hasFileRef = (section) => section
+    .replace(/\s+/g, '') // remove spaces
+    .includes('<!--include')
+
+/* 
+  Takes a string with many includes and normalizes all of the includes so that the contents
+  can be imported or references to files can be created properly.
+  Includes input formats
+     <!-- include: path/to/file -->
+     <!-- include(path/to/file) -->
+     <!-- include(path/to/file -->
+  Includes output format <!-- include(path/to/file) -->
+*/
+module.exports.normalizeIncludes = (str) => str
+    .split('<!-- include')
+    .map((s, i) => {
+        if (i === 0) {
+            return s;
+        }
+        let include = s
+            .substr(0, s.indexOf('-->')) 
+            .trim()
+      
+        if (include.slice(-1) === ')'){
+            include = include.substr(0, include.length - 1)
+        }
+        if (include.slice(0, 1) === '(' || include.slice(0, 1) === ':'){
+            include = include.substr(1, include.length)
+        }
+        const restOfStr = s.substr(s.indexOf('-->') + 3, s.length)         
+        return include.trim() + ') -->' + restOfStr
+    })
+    .join('<!-- include('); 
+
+// Return a reference object to the file path from the include statement.
+// Even though includes should now be normalized, this handles a scenario where they may not be.
+module.exports.getRefFromInclude = (include) => {
+    let path = include.replace(/\s+/g, '') 
+    if (path.includes('include(') && path.includes('-->')) {
+        path = path.substring(
+            path.indexOf('(') + 1,
+            path.indexOf('-->')
+        )
+        // the closing paren is optional so we remove it if it is there.
+        const lastChar = path.slice(-1)
+        if (lastChar === ')'){
+            path = path.substring(0, path.length - 1)
+        }
+    } else if (path.includes('include:') && path.includes('-->')) {
+        path = path.substring(
+            path.indexOf(':') + 1,
+            path.indexOf('-->')
+        )
+    } else {
+        throw Error('Invalid include syntax:' + include)
+    }
+
+    return { $ref: path }
+}
+
+module.exports.searchDataStructure = function (contents, openApi3) {
```

Comment:
Instead of explicitly passing in `openapi3`, let's use the `options` variable or `context` to derive the state

Reply:
@funkeyfreak  Are you suggesting using options globally or simply to pass in options as a parameter _instead_ of openApi3?

Reply:
As you have mentioned, I am suggesting passing in `options` as a parameter rather than a boolean specifying openapi directly. This way, the code remains extendable and not cluttered with conversion-specific logics.